Add .p-stepped-list—detailed

barrymcgee opened the pull request:
## Done

- Added `.p-stepped-list—detailed` pattern

## QA

- Sync code
- Run `gulp jekyll`
- Compare `http://127.0.0.1:4001/vanilla-brochure-theme/examples/patterns/stepped-list-detailed/` with [design spec](https://raw.githubusercontent.com/ubuntudesign/vanilla-brochure-theme-design/master/List%20stepped%20detailed/list-stepped-detailed.png)

Fixes #5

anthonydillon commented on scss/_patterns_stepped-lists.scss:
```diff
@@ -0,0 +1,42 @@
+  .p-stepped-list {
+
+    &--detailed {
+
```
Remove white space please.

barrymcgee replied:
IMO, There should be whitespace here as it's cleaner

Having just did a quick check through Vanilla, it would seem the majority of patterns do but quite a number do not. I'll raise a ticket on Vanilla to correct this inconsistency.

anthonydillon replied:
Thanks, as long as we have a standard. I'm happy.

anthonydillon commented on scss/_patterns_stepped-lists.scss:
```diff
@@ -0,0 +1,42 @@
+        padding: 0 2rem 0 3rem;
+      }
+
+      .list-step__item {
```
Why are we not wrapping the component classes in the pattern namespace? Shouldn't this be `.p-stepped-list__item`?

barrymcgee replied:
Whoever created the original stepped list in Vanilla did not namespace it. I raised a bug in Vanilla about this.

anthonydillon replied:
I think we need to refactor this component then. Can we get that done in this PR? If not we run the risk of a release and then having to maintain these out of date classes.

barrymcgee replied:
Upstream PR- https://github.com/vanilla-framework/vanilla-framework/pull/875

anthonydillon replied:
Thanks

anthonydillon commented on scss/_patterns_stepped-lists.scss:
```diff
@@ -0,0 +1,42 @@
+      list-style: none;
+
+      @media (max-width: $grid-max-width) {
+        padding: 0 2rem 0 3rem;
```
The spec says these values should equal: `0 20px 40px`. They don't currently.

barrymcgee replied:
The spec outlines the expected visual, not the expected code?

anthonydillon replied:
My issue is not the the use of `rems` in this case. It's the fact that the rems used does not produce the expected spec outcomes.

For example 2rem != 20px on small screens.

barrymcgee replied:
@anthonydillon But the spec doesn't specify the code to be used, it outlines the expected visual?

"Vertical space between number/title and content block: 24px"

Screenshot of spec: https://cl.ly/2n3E1Q010q2U
Screenshot of my local: https://cl.ly/072V2n391D3C

anthonydillon replied:
I was looking at the 20px around the content. Sorry you are right, its 24px from number to content.

anthonydillon commented:
![stepped list detailed vanilla brochure theme examples](https://cloud.githubusercontent.com/assets/1413534/22894660/b898054a-f211-11e6-8e21-e203cb08215e.png)

barrymcgee commented:
@anthonydillon The design has this pattern sitting inside a dark strip but I didn't include that here as it's a separate pattern.

anthonydillon commented:
The issue I was seeing was the list-styling was not set to none. As you can see from the screen shot. I can see from the code it should be. Can you confirm your seeing a different thing?

barrymcgee commented:
@anthonydillon Sound like you were on the incorrect branch?

I followed QA steps as outlined above and see: https://cl.ly/3V0g1A2b2v30

anthonydillon commented:
QA issues:
- The title should align vertically with the number in the step
- On large screens, the content box should have 40px padding not 20px
- On large screens, the title should be 24px from the number
- On large screens, the number and title should vertically align with the start of the content on the right
- On large screens, the number bullet is off the screen (see below)

![stepped list detailed vanilla brochure theme examples](https://cloud.githubusercontent.com/assets/1413534/22971966/875c5350-f36f-11e6-9ab2-8f8a61cdf06b.png)

barrymcgee commented:
Aside from the changes needed after the upstream change (vanilla-framework/vanilla-framework#875), this should be ready for re-review again.

anthonydillon commented:
Looks like we have lost some styling with the refactor:

![stepped list detailed vanilla brochure theme examples](https://cloud.githubusercontent.com/assets/1413534/23072442/99fa53a2-f529-11e6-930b-0a0d43da550d.png)

barrymcgee commented:
@anthonydillon Hard refresh?

WillMoggridge commented:
My checkout looks the same as Ant's latest picture.

barrymcgee commented:
@anthonydillon @WillMoggridge Thanks - figured it out.. Vanilla dep was set for `^1.1.4`, which meant you were pulling latest `1.1.7` whereas I was still on `1.1.4`. 

I've now updated the version and updated required class names.
